### ISSUE TITLE
Task has no ports listed but app has two ports configured

### ISSUE THREAD

**Issue by lloesche** [mesosphere/marathon#944]
We have a task for which marathon shows no random ports even though two service ports are configured for the application.

This issue is related to:
https://github.com/mesosphere/servicerouter/issues/1

The WebUI shows:
![screen shot 2015-01-06 at 18 09 11](https://cloud.githubusercontent.com/assets/2124094/5632500/858cb6aa-95cf-11e4-9413-796ad21c8c16.png)

And the /v2/apps definition outputs:

```
{
  "app": {
    "lastTaskFailure": {
      "version": "2014-12-19T01:39:49.443Z",
      "timestamp": "2015-01-06T12:23:24.801Z",
      "taskId": "mm_service_kairosdb-finagle-server.0bfc7995-8a11-11e4-9a92-fe540093056c",
      "state": "TASK_LOST",
      "message": "Reconciliation: Task is unknown",
      "host": "srv3.hw.ca1.mesosphere.com",
      "appId": "/mm/service/kairosdb-finagle-server"
    },
    "tasks": [
      {
        "appId": "/mm/service/kairosdb-finagle-server",
        "version": "1970-01-01T00:00:00.000Z",
        "stagedAt": "2015-01-06T12:23:24.931Z",
        "startedAt": "2015-01-06T17:03:24.934Z",
        "ports": [],
        "host": null,
        "id": "mm_service_kairosdb-finagle-server.5bd9fcc2-9537-11e4-9d9e-56847afe9799"
      },
      {
        "healthCheckResults": [
          {
            "consecutiveFailures": 0,
            "lastSuccess": "2015-01-06T17:06:21.529Z",
            "firstSuccess": "2015-01-06T13:28:42.883Z",
            "taskId": "mm_service_kairosdb-finagle-server.e746c7a6-95a7-11e4-a294-fe5400185893"
          }
        ],
        "appId": "/mm/service/kairosdb-finagle-server",
        "version": "2015-01-06T02:40:56.692Z",
        "stagedAt": "2015-01-06T13:28:30.022Z",
        "startedAt": "2015-01-06T17:03:24.913Z",
        "ports": [
          31525,
          31526
        ],
        "host": "srv2.hw.ca1.mesosphere.com",
        "id": "mm_service_kairosdb-finagle-server.e746c7a6-95a7-11e4-a294-fe5400185893"
      },
      {
        "healthCheckResults": [
          {
            "consecutiveFailures": 0,
            "lastSuccess": "2015-01-06T17:06:21.528Z",
            "firstSuccess": "2015-01-06T13:24:12.702Z",
            "taskId": "mm_service_kairosdb-finagle-server.3e50fee3-95a7-11e4-a294-fe5400185893"
          }
        ],
        "appId": "/mm/service/kairosdb-finagle-server",
        "version": "2015-01-06T02:40:56.692Z",
        "stagedAt": "2015-01-06T13:23:46.554Z",
        "startedAt": "2015-01-06T17:03:24.849Z",
        "ports": [
          31508,
          31509
        ],
        "host": "srv2.hw.ca1.mesosphere.com",
        "id": "mm_service_kairosdb-finagle-server.3e50fee3-95a7-11e4-a294-fe5400185893"
      }
    ],
    "deployments": [],
    "tasksRunning": 3,
    "tasksStaged": 0,
    "version": "2015-01-06T02:40:56.692Z",
    "upgradeStrategy": {
      "minimumHealthCapacity": 1
    },
    "dependencies": [],
    "healthChecks": [
      {
        "maxConsecutiveFailures": 3,
        "timeoutSeconds": 5,
        "intervalSeconds": 30,
        "gracePeriodSeconds": 120,
        "portIndex": 1,
        "protocol": "HTTP",
        "path": "/admin/ping"
      }
    ],
    "container": {
      "docker": {
        "parameters": {},
        "privileged": false,
        "image": "mesosphere/kairosdb-finagle-server:0.9.6-SNAPSHOT-13-master.c794d6963"
      },
      "volumes": [],
      "type": "DOCKER"
    },
    "backoffFactor": 1.15,
    "mem": 5120,
    "cpus": 4,
    "instances": 3,
    "env": {
      "REPFACTOR": "1",
      "CASS_HOSTS": "int.srv2.hw.ca1.mesosphere.com:9160,int.srv3.hw.ca1.mesosphere.com:9160,int.srv4.hw.ca1.mesosphere.com:9160",
      "CORE_COUNT": "4",
      "DOG_STATSD_ENABLED": "true",
      "JAVA_OPTS": "-Xms4g -Xmx4g -XX:PermSize=128m -XX:MaxPermSize=128m -XX:+UnlockExperimentalVMOptions -XX:G1NewSizePercent=50 -XX:G1MaxNewSizePercent=75 -XX:+UseG1GC -XX:MaxGCPauseMillis=20 -XX:InitiatingHeapOccupancyPercent=35"
    },
    "user": null,
    "args": null,
    "cmd": "export PORT_HTTP=$PORT0 && export PORT_HTTP_ADMIN=$PORT1 && /usr/bin/kairos",
    "id": "/mm/service/kairosdb-finagle-server",
    "disk": 0,
    "executor": "",
    "constraints": [
      [
        "hostname",
        "GROUP_BY"
      ]
    ],
    "uris": [],
    "storeUrls": [],
    "ports": [
      9180,
      9190
    ],
    "requirePorts": false,
    "backoffSeconds": 1
  }
}
```

**Comment by kolloch** [comment]
That should indeed be a bug. Does that happen sporadically or deterministically? I didn't really encounter this behaviour yet, so I wonder if it is worth trying to reproduce it.

**Comment by lloesche** [comment]
@kolloch this was resolved in 0.8.0
